system/Mail/MessageInterface.php

```diff
@@ -0,0 +1,315 @@
+<?php namespace CodeIgniter\Mail;
+
+interface MessageInterface
+{
+    /**
+     * Takes an array of options whose keys should match
+     * any of the class properties. The property values will be set.
+     * Any unrecognized elements will be stored in the $data array
+     * to act as view data.
+     *
+     * Example:
+     *
+     *  $message = new App\Mail\UserWelcome([
+     *      'to' => 'John Doe <john.doe@example.com>',
+     *      'from' => 'Jane Doe <jane.doe@example.com>'
+     *  ]);
+     *
+     * @param array|null $options
+     */
+    public function __construct(array $options=null);
```

Comment:
I think interface should not define `__construct()`, we may need an abstract class for `__construct()` requirement.